src/Extensions/ContentReviewCMSExtension.php

```diff
@@ -15,64 +19,127 @@
  */
 class ContentReviewCMSExtension extends LeftAndMainExtension
 {
+    private static $allowed_actions = [
+        'ReviewContentForm',
+        'submitReview',
+    ];
+
     /**
-     * @var array
+     * URL handler for the "content due for review" form
+     *
+     * @param HTTPRequest $request
+     * @return Form|null
      */
-    private static $allowed_actions = array(
-        "savereview"
-    );
+    public function ReviewContentForm(HTTPRequest $request)
+    {
+        // Get ID either from posted back value, or url parameter
+        $id = $request->param('ID') ?: $request->postVar('ID');
+        return $this->getReviewContentForm($id);
+    }
 
     /**
-     * Save the review notes and redirect back to the page edit form.
+     * Return a handler for "content due for review" forms, according to the given object ID
      *
-     * @param array $data
-     * @param Form  $form
-     *
-     * @return string
-     *
-     * @throws HTTPResponse_Exception
+     * @param  int $id
+     * @return Form|null
      */
-    public function savereview($data, Form $form)
+    public function getReviewContentForm($id)
     {
-        $page = $this->findRecord($data);
-        if (!$page->canEdit()) {
-            return Security::permissionFailure($this->owner);
+        $record = SiteTree::get()->byID($id);
+
+        if (!$record) {
+            $this->owner->httpError(404, _t(__CLASS__ . '.ErrorNotFound', 'That object couldn\'t be found'));
+            return null;
         }
 
-        $notes = (!empty($data["ReviewNotes"])
-            ? $data["ReviewNotes"]
-            : _t("ContentReview.NOCOMMENTS", "(no comments)"));
-        $page->addReviewNote(Member::currentUser(), $notes);
-        $page->advanceReviewDate();
+        $user = Security::getCurrentUser();
+        if (!$record->canEdit() || ($record->hasMethod('canBeReviewedBy') && !$record->canBeReviewedBy($user))) {
```

Comment:
When checking this, noted that `ContentReviewLog` doesn't really have a strong permissions model (no canEdit, canView is any logged in user...) perhaps unrelated to this commit but we need to look into model level permission checks